Automattic/woocommerce-payments
# Improve Dev Mode Indicators

Fixes #7498 

#### Changes proposed in this Pull Request









* Updates the style of the Dev Mode indicators on the Settings page:
* Update to use the "Enable live payments" modal previously only used on the Overview page when `Set up payments` is clicked:
* Removed the "Setup Live Payments" on the Overview page since this will now be handled by the dev mode indicators on the Overview and settings pages.
* Update the style of the `TestModeNotice` component, using the new `InlineBanner` component.

Screenshots:

Overview page in Dev Mode:

<img width="911" alt="Screenshot 2023-11-24 at 14 06 08" src="https://github.com/Automattic/woocommerce-payments/assets/11770181/5c2d88ac-6854-4358-bb2b-e992cd252c36">

Overview page in test mode:

<img width="766" alt="Screenshot 2023-11-24 at 14 10 24" src="https://github.com/Automattic/woocommerce-payments/assets/11770181/da4b57e7-1c7d-40b5-8614-729a59cea9c5">

Settings page in test mode:

<img width="1178" alt="Screenshot 2023-11-24 at 10 32 51" src="https://github.com/Automattic/woocommerce-payments/assets/11770181/a5fbafcd-05ca-471a-8111-7634c24531c8">

Settings page in Dev mode:

<img width="1147" alt="Screenshot 2023-11-24 at 10 31 47" src="https://github.com/Automattic/woocommerce-payments/assets/11770181/8f29f772-bc11-4349-b724-469050599d3e">


Modal on Settings Page:

<img width="1195" alt="Screenshot 2023-11-23 at 10 59 08" src="https://github.com/Automattic/woocommerce-payments/assets/11770181/3af00fc2-ae5c-4267-a9da-b34d08ef5767">

Transactions page:

<img width="2374" alt="Screenshot 2023-11-23 at 11 03 08" src="https://github.com/Automattic/woocommerce-payments/assets/11770181/72c069d8-055a-4629-bf57-53ce8e6e38a7">

Transactions details page:

<img width="2373" alt="Screenshot 2023-11-23 at 11 02 19" src="https://github.com/Automattic/woocommerce-payments/assets/11770181/9b7da0b1-4aaf-4e31-bc6a-6f1d3f104f66">

Disputes page:

<img width="2356" alt="Screenshot 2023-11-23 at 11 03 12" src="https://github.com/Automattic/woocommerce-payments/assets/11770181/974bfba4-6d5c-49e9-85f2-d921ed1e14ba">

Disputes details page: 

<img width="2358" alt="Screenshot 2023-11-23 at 11 04 44" src="https://github.com/Automattic/woocommerce-payments/assets/11770181/2afe869b-a7d6-4d24-8a47-01e7a49a8c05">

Deposits page:

<img width="2351" alt="Screenshot 2023-11-23 at 11 03 45" src="https://github.com/Automattic/woocommerce-payments/assets/11770181/9bb6b0cd-4433-426d-b246-d3aef15dd0f0">

#### Testing instructions





* Set up a new JN site with this branch, or locally. Enable Progressive Onboarding (new KYC) in the dev tools, but disable dev mode.
* Onboard a new account in builder mode.  
* Verify that the Overview, Settings and Deposits/Disputes/Transactions pages match the designs - verify the `Learn more` links work correctly.
* Make a test transaction and visit the transaction details page and ensure the test mode notice appears on this page and matches the screenshot above.
* Verify the `Set up payments` button on the Overview page opens the "Setup Live Payments" modal.
* Verify the `Set up payments` button on the Settings page opens the "Setup live payments" modal.
* Use the modal to initiate a full onboarding. This time, onboard a live account.
* After the setup is complete, verify you see a different variant of the test mode notice on the overview page ("WooPayments is in test mode. All transactions will be simulated. Learn more")
* Verify that you can toggle test mode on/off on the settings page, and make a test mode transaction and verify it works.

-------------------


**Post merge**

Update the docs here: https://woo.com/document/woopayments/testing-and-troubleshooting/dev-mode/?quid=e81abdcf1257f23988197cc4883d4dde
